spec/models/payment_confirmation_spec.rb

```diff
@@ -43,6 +43,17 @@
 
       expect(payroll_run.reload.confirmation_report_uploaded_by).to eq(admin_user_id)
     end
+
+    it "queues emails" do
+      travel_to Date.parse("2019-01-01") do
+        friday = Time.parse("2019-01-04").to_i
+        expect { payment_confirmation.ingest }.to have_enqueued_job(ActionMailer::DeliveryJob)
+          .with("ClaimMailer", "payment_confirmation", "deliver_now", payroll_run.claims[0], friday)
```

Comment:
I worry when I see specs like this that they are quite tightly coupled to the internal implementation details of how ActionMailer/ActiveJob works. If that should change, then these specs would break, even though the code itself would probably still work. See the [idea of structure vs behaviour in tests in this blog post](https://medium.com/@kentbeck_7670/test-desiderata-94150638a4b3).

I think these specs would be more robust if they were written to check the actual content of the delivered emails, ala this rejection feature spec: https://github.com/DFE-Digital/dfe-teachers-payment-service/blob/6d12a3d7655c5118d2b937b844ad5530c47ca1ac/spec/features/admin_claim_check_spec.rb#L62

The downside to this is that we will end up duplicating coverage a bit in that this spec will test the email content in a similar way to the specs for the mailer, but I think a small amount of duplication and overlapping coverage in specs is ok if it makes them less structure-sensitive. Here we only need to verify enough to know that the correct date is present in the email body.

Reply:
Yeah, that makes sense. I was a bit iffy about doing it that way at the time, as we're not _strictly_ queuing a job directly when we use `deliver_later`, but I wanted to avoid duplication. I think it's OK as long as we test the bare minimum.

Reply:
yep! 👍